ibis/backends/tests/test_client.py

```diff
@@ -202,3 +202,97 @@ def test_separate_database(con, alternate_current_database, current_data_db):
     db = con.database(current_data_db)
     assert db.name == current_data_db
     assert tmp_db.name == alternate_current_database
+
+
+@pytest.mark.only_on_backends(
+    ['sqlite', 'postgres', 'mysql'],
+    reason="run only if backend is SQLAlchemy based",
+)
+def test_insert_from_dataframe(con):
+
+    sch = ibis.schema(
+        [
+            ('first_name', 'string'),
+            ('last_name', 'string'),
+            ('department_name', 'string'),
+            ('salary', 'float64'),
+        ]
+    )
+
+    df = pd.DataFrame(
+        {
+            'first_name': ['A', 'B', 'C'],
+            'last_name': ['D', 'E', 'F'],
+            'department_name': ['AA', 'BB', 'CC'],
+            'salary': [100.0, 200.0, 300.0],
+        }
+    )
+
+    temp_table = 'temp_to_table'
+    con.drop_table(temp_table, force=True)
+    con.create_table(temp_table, schema=sch)
+    con.load_data(temp_table, df, if_exists='replace')
+    temporary = con.table(temp_table)
+    assert len(temporary.execute()) == 3
```

Comment:
That's really nice. If I'm not wrong, this setup part is the same in the other test. If it is, instead of reusing you can write a pytest fixture (or a test is you need params or makes more sense) that does this and returns the table name (I think that's what it's needed in the test). What do you think?

Reply:
Now, after updating the test case there's no data to be added to the `to_table` so this won't be relevant. But the idea was a good one.